core/chains/evm/client/simulated_backend_client.go

```diff
@@ -708,23 +665,15 @@ func toCallMsg(params map[string]interface{}) ethereum.CallMsg {
 	return callMsg
 }
 
-func (c *SimulatedBackendClient) IsL2() bool {
-	return false
-}
-
-func (c *SimulatedBackendClient) fetchHeader(ctx context.Context, blockNumOrTag string) (*types.Header, error) {
-	switch blockNumOrTag {
-	case rpc.SafeBlockNumber.String():
-		return c.b.Blockchain().CurrentSafeBlock(), nil
-	case rpc.LatestBlockNumber.String():
-		return c.b.Blockchain().CurrentHeader(), nil
-	case rpc.FinalizedBlockNumber.String():
-		return c.b.Blockchain().CurrentFinalBlock(), nil
+func interfaceToAddress(value interface{}) (common.Address, error) {
+	switch v := value.(type) {
+	case common.Address:
+		return v, nil
+	case string:
+		return common.HexToAddress(v), nil
+	case *big.Int:
+		return common.BigToAddress(v), nil
```

Comment:
Are there variants that return errors instead of failing silently?

Reply:
So far, the way it is used in tests, no. And it shouldn't fail silently. Everywhere this function is used will panic if it can't handle the value type. It's not ideal, but the way this is mainly used is in unit testing. So if it fails, the unit tests will fail.

Reply:
The `Hex/BigToAddress` funcs silently accept invalid values that could be difficult to debug from an indirect failure. Why don't we use `IsHexAddress`?
And could check for `big.Int > 0` and `b.Bytes() <=20`?

Reply:
Ahh. Ok. I'll give that a Go...